python/pyspark/sql/context.py

```diff
@@ -75,6 +75,8 @@ class SQLContext(object):
         SQLContext in the JVM, instead we make all calls to this object.
     """
 
+    _instantiatedContext = None
```

Comment:
Note that there are some locks inside of the PySpark SparkContext for handling something similar to this, so it's not a whole lot of work to add it here. In fact, we could just synchronize on `SparkContext._lock` when accessing this field.